packages/vite-plugin/src/index.ts

```diff
@@ -0,0 +1,276 @@
+import type { PluginOption } from 'vite';
+import { hasImportedSideEffects } from '@stylable/build-tools';
+import { resolveConfig as resolveStcConfig, STCBuilder } from '@stylable/cli';
+import type { DiagnosticsMode } from '@stylable/core/dist/index-internal';
+import { emitDiagnostics, tryCollectImportsDeep } from '@stylable/core/dist/index-internal';
+import { resolveNamespace as resolveNamespaceNode } from '@stylable/node';
+import { StylableOptimizer } from '@stylable/optimizer';
+import decache from 'decache';
+import fs from 'fs';
+import { Stylable } from '@stylable/core';
+import {
+    emitAssets,
+    generateCssString,
+    generateStylableModuleCode,
+    getDefaultMode,
+} from './plugin-utils';
+
+export interface StylableVitePluginOptions {
+    optimization?: {
+        minify?: boolean;
+    };
+    inlineAssets?: boolean | ((filepath: string, buffer: Buffer) => boolean);
+    fileName?: string;
+    mode?: 'development' | 'production';
+    diagnosticsMode?: DiagnosticsMode;
+    resolveNamespace?: typeof resolveNamespaceNode;
+    /**
+     * Runs "stc" programmatically with the webpack compilation.
+     * true - it will automatically detect the closest "stylable.config.js" file and use it.
+     * string - it will use the provided string as the "stcConfig" file path.
+     */
+    stcConfig?: boolean | string;
+    projectRoot?: string;
```

Comment:
the rollup integration this was copied from now supports a better default for handling side-effects: `includeGlobalSideEffects`